File: benchmarks/go/11_string_concat_scan.go
// String Concat and Scan Benchmark - Go Implementation

package main

import (
	"fmt"
	"strconv"
	"strings"
)

func main() {
	var builder strings.Builder
	builder.Grow(4096)

	for i := 0; i < 250; i++ {
		builder.WriteString("item=")
		builder.WriteString(strconv.Itoa(i))
		builder.WriteByte(';')
	}

	text := builder.String()
	var digits, equals, semicolons int64
	for _, ch := range text {
		if ch >= '0' && ch <= '9' {
			digits++
		} else if ch == '=' {
			equals++
		} else if ch == ';' {
			semicolons++
		}
	}

	fmt.Printf("string concat checksum: %d\n", int64(len(text))+digits+equals+semicolons)
}

File: benchmarks/rust/11_string_concat_scan.rs
// String Concat and Scan Benchmark - Rust Implementation

use std::fmt::Write;

fn main() {
    let n = 250;
    let mut text = String::with_capacity(4096);

    for i in 0..n {
        let _ = write!(&mut text, "item={};", i);
    }

    let mut digits = 0i64;
    let mut equals = 0i64;
    let mut semicolons = 0i64;
    for ch in text.chars() {
        if ch.is_ascii_digit() {
            digits += 1;
        } else if ch == '=' {
            equals += 1;
        } else if ch == ';' {
            semicolons += 1;
        }
    }

    println!(
        "string concat checksum: {}",
        text.len() as i64 + digits + equals + semicolons
    );
}
